synthesizer/process/src/cost.rs

```diff
@@ -60,7 +60,14 @@ pub fn deployment_cost<N: Network>(deployment: &Deployment<N>) -> Result<(u64, (
 /// Returns the *minimum* cost in microcredits to publish the given execution (total cost, (storage cost, finalize cost)).
 pub fn execution_cost<N: Network>(process: &Process<N>, execution: &Execution<N>) -> Result<(u64, (u64, u64))> {
     // Compute the storage cost in microcredits.
-    let storage_cost = execution.size_in_bytes()?;
+    let mut storage_cost = execution.size_in_bytes()?;
+
+    // Compute a quadratic storage cost penalty if above the size penalty threshold.
+    if storage_cost > N::EXECUTION_STORAGE_PENALTY_THRESHOLD {
+        storage_cost = storage_cost
+            .saturating_pow(N::EXECUTION_STORAGE_FEE_DEGREE)
```

Comment:
I strongly advise we use a quadratic function otherwise use an approximation algorithm for `pow`, similar to how to do `retarget` in the puzzle.

Reply:
May I ask how this would be non-determinism here? Both `storage_cost` and the fee degree are integers which power in rust is deterministic for.

And also what is meant for other languages? WASM compatiably?